patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1029,39 +1029,117 @@
 +    {
 +        return this == net.minecraft.init.Blocks.field_150475_bE || this == net.minecraft.init.Blocks.field_150340_R || this == net.minecraft.init.Blocks.field_150484_ah || this == net.minecraft.init.Blocks.field_150339_S;
 +    }
-+
++    
 +    /**
-+     * Rotate the block. For vanilla blocks this rotates around the axis passed in (generally, it should be the "face" that was hit).
-+     * Note: for mod blocks, this is up to the block and modder to decide. It is not mandated that it be a rotation around the
-+     * face, but could be a rotation to orient *to* that face, or a visiting of possible rotations.
-+     * The method should return true if the rotation was successful though.
++     * Checks if the block can be rotated around the specified axis, clockwise as looking from that side. Doesn't take into
++     * account the player and where on the block they're looking.<br/>
++     * Note: for mod blocks, this is up to the block and modder to decide, but it <b>IS</b> mandated that it be a rotation around the
++     * face, for compatibility reasons. If you know about the player and where they're looking at, implement
++     * {@link Block#canPlayerRotate(World, BlockPos, EnumFacing, EntityPlayer, RayTraceResult)}.
 +     *
 +     * @param world The world
 +     * @param pos Block position in world
 +     * @param axis The axis to rotate around
-+     * @return True if the rotation was successful, False if the rotation failed, or is not possible
++     * @return True if it can be rotated, false if it can't
 +     */
-+    public boolean rotateBlock(World world, BlockPos pos, EnumFacing axis)
++    public boolean canRotateBlockAround(World world, BlockPos pos, EnumFacing axis)
```

Comment:
Why do we need a can function? we specificlaly removed that crap when we simplified the rotation.